lib/IRGen/ClassLayout.h

```diff
@@ -96,14 +136,23 @@ class ClassLayout {
 
   bool isFixedLayout() const { return IsFixedLayout; }
 
-  bool isFixedSize() const { return IsFixedSize; }
+  bool isFixedSize() const {
+    return !(Options.contains(ClassMetadataFlags::ClassHasMissingMembers) ||
+             Options.contains(ClassMetadataFlags::ClassHasResilientMembers) ||
+             Options.contains(ClassMetadataFlags::ClassHasGenericLayout) ||
+             Options.contains(ClassMetadataFlags::ClassHasObjCAncestry));
```

Comment:
If ancestry matters, shouldn't `ClassHasResilientAncestry` be included here too? If it doesn't, shouldn't `ClassHasObjCAncestry` be omitted?

(Please add test cases for this distinction too.)

Reply:
If "resilient ancestry" is true but "resilient members" is not true, you've inherited from a @_fixed_layout class in another module. This means you have to instantiate your metadata, but you still have a fixed size _instance_ whose stored property layout is fully known. I'll add a comment explaining what isFixedSize() is used for. I added tests for this case when I redid what it means to have a @_fixed_layout class.

Reply:
Oof, I keep forgetting that the "members" flag includes ancestors.